x-pack/legacy/plugins/maps/public/embeddable/map_embeddable.js

```diff
@@ -176,37 +171,38 @@ export class MapEmbeddable extends Embeddable {
   }
 
   _handleStoreChanges() {
-    let embeddableConfigChanged = false;
 
     const center = getMapCenter(this._store.getState());
     const zoom = getMapZoom(this._store.getState());
-    if (!this._embeddableConfig.mapCenter
-      || this._embeddableConfig.mapCenter.lat !== center.lat
-      || this._embeddableConfig.mapCenter.lon !== center.lon
-      || this._embeddableConfig.mapCenter.zoom !== zoom) {
-      embeddableConfigChanged = true;
-      this._embeddableConfig.mapCenter = {
-        lat: center.lat,
-        lon: center.lon,
-        zoom: zoom,
-      };
+
+
+    const mapCenter = this.input.mapCenter || {};
+    if (!mapCenter
+      || mapCenter.lat !== center.lat
+      || mapCenter.lon !== center.lon
+      || mapCenter.zoom !== zoom) {
+      this.updateInput({
+        mapCenter: {
+          lat: center.lat,
+          lon: center.lon,
+          zoom: zoom,
+        }
+      });
     }
 
     const isLayerTOCOpen = getIsLayerTOCOpen(this._store.getState());
-    if (this._embeddableConfig.isLayerTOCOpen !== isLayerTOCOpen) {
-      embeddableConfigChanged = true;
-      this._embeddableConfig.isLayerTOCOpen = isLayerTOCOpen;
-    }
 
-    const openTOCDetails = getOpenTOCDetails(this._store.getState());
-    if (!_.isEqual(this._embeddableConfig.openTOCDetails, openTOCDetails)) {
-      embeddableConfigChanged = true;
-      this._embeddableConfig.openTOCDetails = openTOCDetails;
+    if (!this.input.isLayerTOCOpen
```

Comment:
`!this.input.isLayerTOCOpen` is not needed in this check and will fire updateInput every time input. isLayerTOCOpen is false.

Should just be
`if (this.input.isLayerTOCOpen !== isLayerTOCOpen) `